components/resources/library/src/webMain/kotlin/org/jetbrains/compose/resources/Resource.web.kt

```diff
@@ -50,4 +57,39 @@ internal fun getResourceUrl(windowOrigin: String, windowPathname: String, resour
         path.startsWith("http://") || path.startsWith("https://") -> path
         else -> windowOrigin + windowPathname + path
     }
+}
+
+@ExperimentalResourceApi
+@Composable
+fun preloadAndCacheFont(
```

Comment:
As we agreed:
 - add documentation
 - rename to `preloadFont`
 - check environment changes

Reply:
done. 
Added a couple of tests. Fixed rememberResourceState on web to not ignore the environment changes.

The `preload*` funcitons now react on evnironment changes. It will switch to a default font or default bitmap after the environment changes if a new env has corresponding resources. It means that the progress indicator will be shown again (in our demo).